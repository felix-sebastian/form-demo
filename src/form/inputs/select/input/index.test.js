import React from "react";
import Pill from ".";
import renderer from "react-test-renderer";

it("displays the current value", () => {
  const result = renderer
    .create(
      <Pill
        value="Foo"
        options={["Foo", "Bar", "Baz"]}
        cursor={0}
        query="Foo"
      />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});

it("displays the cursored option", () => {
  const result = renderer
    .create(
      <Pill value="Foo" focused options={["Foo", "Bar", "Baz"]} cursor={1} />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});
