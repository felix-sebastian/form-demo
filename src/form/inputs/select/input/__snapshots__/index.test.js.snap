// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the current value 1`] = `
<div
  className="sc-AykKC KLKSr"
  data-testid="input"
>
  <input
    className="sc-AykKD hcLsZH sc-AykKE ddfZRt"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Select one"
    value="Foo"
  />
  <div
    className="sc-AykKF chmxPI"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </div>
</div>
`;

exports[`displays the cursored option 1`] = `
<div
  className="sc-AykKC bftxTE"
  data-testid="input"
>
  <input
    className="sc-AykKD hcLsZH sc-AykKE ddfZRt"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Bar"
    value=""
  />
  <div
    className="sc-AykKF chmxPI"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </div>
</div>
`;
